src/Microsoft.TemplateSearch.TemplateDiscovery/Program.cs

```diff
@@ -56,7 +56,7 @@ static void Main(string[] args)
                 throw new NotImplementedException("no checker for the input options");
             }
 
-            PackSourceCheckResult checkResults = packSourceChecker.CheckPackages();
+            PackSourceCheckResult checkResults = packSourceChecker.CheckPackagesAsync().GetAwaiter().GetResult();
```

Comment:
Here you will probably want to do `await` instead of `.GetAwaiter().GetResult();` and change `void Main(string[] args)` into `async Task Main(string[] args)`